src/Components/ServiceScene/Breakdowns/ByFrameRepeater.tsx

```diff
@@ -46,8 +51,24 @@ export class ByFrameRepeater extends SceneObjectBase<ByFrameRepeaterState> {
     }
 
     this.state.body.setState({ children: newChildren });
+    this.unfilteredChildren = newChildren;
   }
 
+  public filterFrame = (filterFn: FrameFilterCallback) => {
+    const data = sceneGraph.getData(this).state.data;
+    if (!data) {
+      return;
+    }
+    const newChildren: SceneFlexItem[] = [];
+    for (let seriesIndex = 0; seriesIndex < data.series.length; seriesIndex++) {
+      if (filterFn(data.series[seriesIndex])) {
+        newChildren.push(this.unfilteredChildren[seriesIndex]);
```

Comment:
@gtk-grafana This is the trick I used to prevent flickering while filtering (similar to what I noticed in patterns). Just prevent re-building panels that should remain visible.